<a href="https://colab.research.google.com/github/KavyaChoudhary2008/class114/blob/main/class114.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import plotly.express as px
df=pd.read_csv("data.csv")
height=df["Height"].tolist()
weight=df["Weight"].tolist()
fig=px.scatter(x=height,y=weight)
fig.show()

The scatter plot is a straight line and we observe as x increases y also increases. And both height and weight are related to each other. The more the hight the more the weight.
This line has a formula: 
 y=mx+c
 m is the slope
 c is the intercept
 we already know the values of x and y as it is weight and height.
 hit and trial method.
 lets assume the value of m and c.

In [ ]:
m=1 
c=0
y=[]
for x in height:
  y_value=m*x + c
  y.append(y_value)

fig=px.scatter(x=height,y=weight)
fig.update_layout(shapes=[ dict( type= 'line', y0= min(y), y1= max(y), x0= min(height), x1= max(height) ) ])
fig.show()

In [ ]:
m=0.95
c=-93
y=[]
for x in height:
  y_value=m*x + c
  y.append(y_value)

fig=px.scatter(x=height,y=weight)
fig.update_layout(shapes=[ dict( type= 'line', y0= min(y), y1= max(y), x0= min(height), x1= max(height) ) ])
fig.show()

We have a close enough line that covers all the scatter plot points in out chart.
Here m is the slope of the line and c is the intercept.

In [ ]:
#Let's predict the value of weight if the height is 250.
x=250
y=m*x + c
print(f"weight of someone with height {x} is {y}")

weight of someone with height 250 is 144.5


lets find the slope and intercept of the data using computer algorithym.

In [ ]:
import numpy as np
height_array=np.array(height)
weight_array=np.array(weight)
#slope and intercept using the formula
m,c=np.polyfit(height_array,weight_array,1)
y=[]
for x in height_array:
  y_value=m*x + c
  y.append(y_value)

fig=px.scatter(x=height_array,y=weight_array)
fig.update_layout(shapes=[ dict( type= 'line', y0= min(y), y1= max(y), x0= min(height_array), x1= max(height_array) ) ])
fig.show()

print(m,c)

0.9600183750530678 -95.66248461433146


In [ ]:
#predict the weight of a person if the height is 250.
x=250
y=m*x + c
print(f"weight of someone with height {x} is {y}")

weight of someone with height 250 is 144.3421091489355


Earlier by putting the values of m and c by hit and trial method we got the weight 144.5 but after finding out the perfect value of m and c by numpy the weight of the person is 144.34.


This is linear regression.